api/groups/rawGroup.go

```diff
@@ -0,0 +1,537 @@
+package groups
+
+import (
+	"fmt"
+	"net/http"
+	"sync"
+	"time"
+
+	"github.com/ElrondNetwork/elrond-go-core/core/check"
+	"github.com/ElrondNetwork/elrond-go-core/data/block"
+	"github.com/ElrondNetwork/elrond-go/api/errors"
+	"github.com/ElrondNetwork/elrond-go/api/shared"
+	"github.com/gin-gonic/gin"
+)
+
+const (
+	getRawMetaBlockByNoncePath       = "/raw/metablock/by-nonce/:nonce"
+	getRawMetaBlockByHashPath        = "/raw/metablock/by-hash/:hash"
+	getRawMetaBlockByRoundPath       = "/raw/metablock/by-round/:round"
+	getRawShardBlockByNoncePath      = "/raw/shardblock/by-nonce/:nonce"
+	getRawShardBlockByHashPath       = "/raw/shardblock/by-hash/:hash"
+	getRawShardBlockByRoundPath      = "/raw/shardblock/by-round/:round"
+	getInternalMetaBlockByNoncePath  = "/json/metablock/by-nonce/:nonce"
+	getInternalMetaBlockByHashPath   = "/json/metablock/by-hash/:hash"
+	getInternalMetaBlockByRoundPath  = "/json/metablock/by-round/:round"
+	getInternalShardBlockByNoncePath = "/json/shardblock/by-nonce/:nonce"
+	getInternalShardBlockByHashPath  = "/json/shardblock/by-hash/:hash"
+	getInternalShardBlockByRoundPath = "/json/shardblock/by-round/:round"
+	getRawMiniBlockByHashPath        = "/raw/miniblock/by-hash/:hash"
+	getInternalMiniBlockByHashPath   = "/json/miniblock/by-hash/:hash"
+)
+
+// rawBlockFacadeHandler defines the methods to be implemented by a facade for handling block requests
+type rawBlockFacadeHandler interface {
+	GetRawMetaBlockByHash(hash string) ([]byte, error)
+	GetRawMetaBlockByNonce(nonce uint64) ([]byte, error)
+	GetRawMetaBlockByRound(round uint64) ([]byte, error)
+	GetRawShardBlockByHash(hash string) ([]byte, error)
+	GetRawShardBlockByNonce(nonce uint64) ([]byte, error)
+	GetRawShardBlockByRound(round uint64) ([]byte, error)
+	GetInternalMetaBlockByHash(hash string) (*block.MetaBlock, error)
```

Comment:
no need for 2 functions. now you have (for example):
`GetRawMetaBlockByNonce(nonce uint64) ([]byte, error)`
and
`GetInternalMetaBlockByNonce(nonce uint64) (*block.MetaBlock, error)`

while you could have had only 1:

`GetInternalMetaBlockByNonce(format outportFormat, nonce uint64) (interface{}, error)`
where:
1. `outportFormat` could be `type outportFormat uint8` and there are 2 possible values:
```
const (
    internal outputFormat = 0 // returns struct directly, will be serialized into JSON by gin
    proto     outputFormat = 1 // returns the bytes of the proto object
)
```
2. when returning the object, if the `outputFormat` is `internal` then return directly the meta block ; and if it is proto, then return the proto bytes
=====
afterwards, on API level you'll only have `shared.RespondWith(c, http.StatusOK, gin.H{"block": block}, "", shared.ReturnCodeSuccess)` where `block` is the `interface{}` that was returned by the facade

Reply:
apply this for all items :pray: